apps/web/pages/api/book/setLocation.ts

```diff
@@ -0,0 +1,168 @@
+import dayjs from "dayjs";
+import { NextApiRequest, NextApiResponse } from "next";
+
+import EventManager from "@calcom/core/EventManager";
+import logger from "@calcom/lib/logger";
+import { getTranslation } from "@calcom/lib/server/i18n";
+import { Prisma } from "@calcom/prisma/client";
+import type { CalendarEvent } from "@calcom/types/Calendar";
+import type { AdditionInformation } from "@calcom/types/Calendar";
+
+import { getSession } from "@lib/auth";
+import { sendLocationChangeEmails } from "@lib/emails/email-manager";
+import prisma from "@lib/prisma";
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method !== "POST") {
+    return res.status(405).json({ message: "Method not allowed" });
+  }
+
+  const bookingId = req.body.bookingId;
+  const location = req.body.newLocation;
+
+  const session = await getSession({ req: req });
+
+  if (!session?.user?.id) {
+    res.status(401).json({ message: "Not authenticated" });
+    return;
+  }
+
+  try {
+    const currentUser = await prisma.user.findFirst({
+      where: {
+        id: session.user.id,
+      },
+      select: {
+        id: true,
+        credentials: {
+          orderBy: { id: "desc" as Prisma.SortOrder },
```

Comment:
any particular reason for the sort on credentials?

Reply:
Just tried it without sorting, and seems like it's not needed. I removed it